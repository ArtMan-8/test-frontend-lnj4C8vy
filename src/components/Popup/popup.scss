.popup-wrapper {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 50;

  display: flex;
  justify-content: center;
  align-items: center;

  background-color: #b3b3b3;
  cursor: pointer;
}

.popup {
  position: absolute;
  z-index: 100;

  display: flex;
  flex-direction: column;
  align-items: center;
  box-sizing: border-box;
  width: 100%;
  max-width: 450px;
  height: 100vh;
  padding: 32px 24px;
  overflow-y: auto;

  background-color: var(--color-white);
  cursor: auto;

  @media(min-width: 768px) {
    height: unset;
    max-height: 100vh;
    padding: 32px;

    border-radius: 30px;
  }
}

.popup__salary {
  margin: 0;
  padding: 0;

  border: none;
}

.popup__legend {
  margin-bottom: 20px;

  font-weight: 500;
  font-size: 18px;

  @media(min-width: 768px) {
    margin-bottom: 22px;

    font-size: 28px;
  }
}

.popup__description {
  margin: 0;

  font-size: 12px;
  line-height: 16px;
  color: var(--color-grey-dark);

  @media(min-width: 768px) {
    font-size: 14px;
    line-height: 24px;
  }
}

.popup__data {
  position: relative;

  margin-top: 26px;
  margin-bottom: 12px;
}

.popup__data-label {
  display: inline-block;
  margin-bottom: 12px;

  font-weight: 500;
  font-size: 14px;
}

.popup__data-input {
  box-sizing: border-box;
  width: 100%;
  padding: 10px;

  color: var(--color-grey);

  border: 1px solid var(--color-grey);
  border-radius: 3px;

  &:hover {
    border-color: var(--color-black);
  }

  &:focus {
    color: var(--color-black);
  }

  &:disabled {
    color: var(--color-grey-dark);

    border-color: var(--color-grey-dark);
  }

  &:valid {
    color: var(--color-black);
  }

  &:invalid {
    border-color: var(--color-red);
    outline: none;
    box-shadow: none;

    & + .popup__data-none {
      position: absolute;
      bottom: -24px;
      left: 0;

      display: inline-block;

      font-size: 10px;
      color: var(--color-red);
    }
  }
}

.popup__data-none {
  display: none;
}

.popup__btn-close {
  position: absolute;
  top: 20px;
  right: 12px;

  width: 20px;
  height: 20px;

  font-size: 0;

  background: transparent;
  border: none;
  cursor: pointer;

  @media(min-width: 768px) {
    top: 22px;
    right: 20px;

    width: 24px;
    height: 24px;
  }

  &:before,
  &:after {
    content: "";
    position: absolute;
    top: 0;

    width: 3px;
    height: 16px;

    background: var(--color-red);
    border-radius: 100%;

    @media(min-width: 768px) {
      width: 4px;
      height: 24px;
    }
  }

  &:before {
    transform: rotate(45deg);
  }

  &:after {
    transform: rotate(-45deg);
  }
}
