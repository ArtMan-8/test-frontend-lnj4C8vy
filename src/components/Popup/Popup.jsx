import React, { useState } from 'react';
import T from 'prop-types';
import './popup.scss';
import CalcDed from '../CalcDed';

const Popup = ({ handleClosePopup }) => {
  const [salary, setSalary] = useState('');

  const salaryFormat = (inputVal) => {
    // const rub = '₽';
    const formatNum = inputVal.toString().replace(/(\d)(?=(\d{3})+$)/g, '$1 ');
    const formatValue = `${formatNum}`;
    return formatValue;
  };

  const salaryInit = (formatVal) => parseInt(formatVal.replace(/\s+/g, '').trim(), 10);

  return (
    <form className="popup">
      <fieldset className="popup__salary">
        <legend className="popup__legend">Налоговый вычет</legend>
        <p className="popup__description">
          Используйте налоговый вычет чтобы погасить ипотеку досрочно.{' '}
          Размер налогового вычета составляет не более 13%{' '}
          от своего официального годового дохода.
        </p>

        <div className="popup__data">
          <label
            htmlFor="salary"
            className="popup__data-label"
          >Ваша зарплата в месяц</label>

          <input
            type="text"
            id="salary"
            className="popup__data-input"
            placeholder="Введите данные"
            required
            value={salaryFormat(salary)}
            onChange={(evt) => {
              const inputValue = salaryInit(evt.target.value);

              if (Number(inputValue)) {
                setSalary(inputValue);
                return;
              }

              setSalary('');
            }}
          />

          <p className="popup__data-none">Поле обязательно для заполнения</p>
        </div>
      </fieldset>

      <CalcDed salary={salary} />

      <button
        type="submit"
        className="popup__btn-add"
        onClick={(evt) => evt.preventDefault()}
      >Добавить</button>

      <button
        type="button"
        className="popup__btn-close"
        onClick={handleClosePopup}
      >Закрыть попап</button>
    </form>
  );
};

Popup.propTypes = {
  handleClosePopup: T.func,
};

export default Popup;
