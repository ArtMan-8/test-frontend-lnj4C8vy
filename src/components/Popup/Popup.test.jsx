import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Popup from './Popup.jsx';

afterEach(cleanup);

test('App snapshot', () => {
  const { asFragment } = render(<Popup />);
  expect(asFragment()).toMatchSnapshot();
});
