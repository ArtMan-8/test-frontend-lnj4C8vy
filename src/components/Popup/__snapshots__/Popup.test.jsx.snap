// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
<DocumentFragment>
  <div
    class="popup-wrapper"
  >
    <form
      class="popup"
    >
      <fieldset
        class="popup__salary"
      >
        <legend
          class="popup__legend"
        >
          Налоговый вычет
        </legend>
        <p
          class="popup__description"
        >
          Используйте налоговый вычет чтобы погасить ипотеку досрочно. Размер налогового вычета составляет не более 13% от своего официального годового дохода.
        </p>
        <div
          class="popup__data"
        >
          <label
            class="popup__data-label"
            for="salary"
          >
            Ваша зарплата в месяц
          </label>
          <input
            class="popup__data-input"
            id="salary"
            placeholder="Введите данные"
            required=""
            type="text"
            value=""
          />
          <p
            class="popup__data-none"
          >
            Поле обязательно для заполнения
          </p>
        </div>
      </fieldset>
      <fieldset
        class="calc-ded"
      >
        <button
          class="calc-ded__calc"
          type="button"
        >
          Рассчитать
        </button>
        <div
          class="calc-ded__controls"
        >
          <h3
            class="calc-ded__controls-header"
          >
            Что уменьшаем?
          </h3>
          <button
            class="calc-ded__controls-btn calc-ded__controls-btn--active"
            type="button"
          >
            Платёж
          </button>
          <button
            class="calc-ded__controls-btn"
            type="button"
          >
            Срок
          </button>
        </div>
        <button
          class="calc-ded__btn-add"
          type="submit"
        >
          Добавить
        </button>
      </fieldset>
      <button
        class="popup__btn-close"
        type="button"
      >
        Закрыть попап
      </button>
    </form>
  </div>
</DocumentFragment>
`;
